### PR TITLE
Idaho subtraction for allowable state childcare expenses

### LINKED ISSUE #3110: Idaho subtraction for allowable state childcare expenses
Skipping from #2935 since it depends on the CDCC, see https://github.com/PolicyEngine/policyengine-us/pull/2935#discussion_r1341482445

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #3110

### CODE REVIEW

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/dependent_care_expenses/cap.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  reference:
+  - title: Idaho Statutes 63-3022D 
+    href: https://legislature.idaho.gov/statutesrules/idstat/title63/t63ch30/sect63-3022d/
+  - title: Idaho Statutes 63-3022D # prior to 2023
+    href: https://law.justia.com/codes/idaho/2022/title-63/chapter-30/section-63-3022d/
+  - title: 2023 Idaho Tax Form 40 Instructions
+    href: https://tax.idaho.gov/wp-content/uploads/forms/EIN00046/EIN00046_11-06-2023.pdf#page=2
+  # The cap was introduced in 2022
+  - title: 2022 Idaho Tax Form 40 Instructions
+    href: https://tax.idaho.gov/wp-content/uploads/forms/EIN00046/EIN00046_11-15-2021.pdf#31
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/dependent_care_expenses/cap.yaml:18]:
```suggestion
    href: https://tax.idaho.gov/wp-content/uploads/forms/EIN00046/EIN00046_11-15-2021.pdf#page=31
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Could you add an integration test going off the worksheet?

**Comment by PavelMakarchuk** [comment]
> Could you add an integration test going off the worksheet?

I see, refactored the relevant cdcc expenses to math the form calc

File: policyengine_us/variables/gov/states/id/tax/income/subtractions/dependent_care_exepenses/id_household_and_dependent_care_expense_deduction.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        expenses = tax_unit("tax_unit_childcare_expenses", period)
+        # In 2023 Idaho implemented an increased cap independent of the number of children
+        if p.available:
+            amount_per_child = p.amount
+        else:
+            capped_count_cdcc_eligible = tax_unit(
+                "capped_count_cdcc_eligible", period
+            )
+            cdcc = parameters(period).gov.irs.credits.cdcc
+            amount_per_child = cdcc.max * capped_count_cdcc_eligible
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/id/tax/income/subtractions/dependent_care_exepenses/id_household_and_dependent_care_expense_deduction.py:25]: rename variable to reflect that it's multiplied by number of children

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/dependent_care_expenses/cap/available.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/dependent_care_expenses/cap/available.yaml:1]: rename `applies.yaml`? though i'm not clear on how this works

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/dependent_care_expenses/cap/available.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+description: Idaho caps the household and dependent care expense deduction independently of the federal CDCC calucaltion if this is true.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/dependent_care_expenses/cap/available.yaml:1]:
```suggestion
description: Idaho caps the household and dependent care expense deduction independently of the federal child and dependent care calculation if this is true.
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Could you describe how this works more?

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/dependent_care_expenses/cap/available.yaml
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/id/tax/income/deductions/dependent_care_expenses/cap/available.yaml:1]: In 2023 Idaho removed the 3,000 for one and 6,000 for two or more children cap on the expenses with a flat 12,000 cap

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Could you refactor `cdcc_relevant_expenses` by creating a new `cdcc_limit` variable, which equals `capped_count_cdcc_eligible * amount`?

Then you can avoid pulling in IRS parameters, remove `p.applies`, turn `p.cap.amount` into `p.cap` (remove folder), and simplify this formula to
```python
limit = max_(tax_unit("cdcc_limit", period), p.cap)
```

**Comment by PavelMakarchuk** [comment]
> Could you refactor `cdcc_relevant_expenses` by creating a new `cdcc_limit` variable, which equals `capped_count_cdcc_eligible * amount`?
> 
> Then you can avoid pulling in IRS parameters, remove `p.applies`, turn `p.cap.amount` into `p.cap` (remove folder), and simplify this formula to
> 
> ```python
> limit = max_(tax_unit("cdcc_limit", period), p.cap)
> ```

I have refactored the cdcc but it forces me to set the cap to 0 pre 2023 to avoid making the $6,000 obsolete

File: policyengine_us/variables/gov/irs/credits/cdcc/cdcc_limit.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    label = "CDCC-relevant care expense limit"
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "https://www.law.cornell.edu/uscode/text/26/21#c",
+        "https://www.law.cornell.edu/uscode/text/26/21#d_1",
+    )
+
+    def formula(tax_unit, period, parameters):
+        cdcc = parameters(period).gov.irs.credits.cdcc
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/irs/credits/cdcc/cdcc_limit.py:16]:
```suggestion
        p = parameters(period).gov.irs.credits.cdcc
```

File: policyengine_us/variables/gov/irs/credits/cdcc/cdcc_limit.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+class cdcc_limit(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "CDCC-relevant care expense limit"
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "https://www.law.cornell.edu/uscode/text/26/21#c",
+        "https://www.law.cornell.edu/uscode/text/26/21#d_1",
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/irs/credits/cdcc/cdcc_limit.py:12]: remove - only c

File: policyengine_us/tests/policy/baseline/gov/irs/credits/cdcc/cdcc_limit.yaml
Comment (MaxGhenis) [inline policyengine_us/tests/policy/baseline/gov/irs/credits/cdcc/cdcc_limit.yaml:1]: do you need to adjust the cdcc_relevant_expenses tests too?

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/dependent_care_expenses/cap.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+description: Idaho caps the household and dependent care expense deduction at this amount.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/dependent_care_expenses/cap.yaml:1]:
```suggestion
description: Idaho caps the household and dependent care expense deduction at the greater of the federal cap and this amount.
```

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/dependent_care_expenses/cap.yaml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  period: year
+  label: Idaho dependent care expense deduction cap amount
+  reference:
+  - title: Idaho Statutes 63-3022D 
+    href: https://legislature.idaho.gov/statutesrules/idstat/title63/t63ch30/sect63-3022d/
+  - title: Idaho Statutes 63-3022D # prior to 2023
+    href: https://law.justia.com/codes/idaho/2022/title-63/chapter-30/section-63-3022d/
+  - title: 2023 Idaho Tax Form 40 Instructions
+    href: https://tax.idaho.gov/wp-content/uploads/forms/EIN00046/EIN00046_11-06-2023.pdf#page=2
+  # The cap was introduced in 2022
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/dependent_care_expenses/cap.yaml:16]:
```suggestion
  # Idaho introduced the higher cap in 2022.
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/tests/policy/baseline/gov/irs/credits/cdcc/cdcc_limit.yaml
Comment (PavelMakarchuk) [reply policyengine_us/tests/policy/baseline/gov/irs/credits/cdcc/cdcc_limit.yaml:1]: seems like we never had a unit test for `cdcc_relevant_expenses`  - added

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**